pingcap/tidb
# executor: add diagnosis rule to check some metrics exceed thresholds

Signed-off-by: crazycs <crazycs520@gmail.com>



### What problem does this PR solve? 
This PR adds a new diagnosis rule -- `threshold-check`, which is used to check some metrics threshold value, like tikv thread cpu usage. We will detect the following metrics tables in the current implementation:

* `tikv_thread_cpu`

Consider the PR length, this PR only add some part checks of `threshold-check` rule, I will add more check in the next PR.

### What is changed and how it works?


### Check List 

Tests 

 - Unit test
 - Manual test 

Release note

 - Add diagnosis rule `threshold-check` which is used to check some metrics threshold value.
